library/data/tags/src/main/java/org/quiltmc/qsl/tag/api/TagRegistry.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC, 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.tag.api;
+
+import java.util.function.Supplier;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.block.Block;
+import net.minecraft.entity.EntityType;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.item.Item;
+import net.minecraft.tag.BlockTags;
+import net.minecraft.tag.EntityTypeTags;
+import net.minecraft.tag.FluidTags;
+import net.minecraft.tag.GameEventTags;
+import net.minecraft.tag.ItemTags;
+import net.minecraft.tag.Tag;
+import net.minecraft.tag.TagGroup;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryKey;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.event.GameEvent;
+
+import org.quiltmc.qsl.tag.impl.TagRegistryImpl;
+
+/**
+ * Represents a tag registry.
+ *
+ * @param <T> the type of the content in the registered tags
+ */
+@ApiStatus.NonExtendable
+public interface TagRegistry<T> {
```

Comment:
I feel it would be useful to have a way to iterate over the tags in a tag registry. Or where ever it is applicable.

Reply:
Since the `TagRegistry` does not keep track of registered tags, that feels a bit impossible?